packages/odata-common/src/request-builder/batch/batch-request-serializer.ts

```diff
@@ -75,6 +88,60 @@ export function serializeRequest(
   ].join(unixEOL);
 }
 
+type AllBuilderTypes<
+  EntityT extends EntityBase,
+  DeSerializersT extends DeSerializers,
+  ReturnT,
+  ParametersT
+> =
+  | Omit<MethodRequestBuilder, 'execute'>
+  | ActionFunctionImportRequestBuilderBase<
```

Comment:
[pp] You could create a type `RequestBuilderWithBatchReference` containing the Create,Delte and so on. Then you define the `AllBuilderType` as `RequestBuilderWithBatchReference | Omit<MethodRequestBuilder, 'execute'>` Then you have the long type only once.